src/adapters/adapter.js

```diff
@@ -40,6 +69,9 @@ class Adapter {
 
             // We're done
             if (item === undefined) {
+              this._setCacheTime(STORE_NAME, CACHE_KEY);
+              globalCache.put(CACHE_KEY, new Response(JSON.stringify(folders[''])));
```

Comment:
When the "Load entire tree at once" is unchecked (`opts.node` is undefined) , the cache shouldn't be used. Maybe should handle that case?

Reply:
Also, maybe we should provide an option to users, like "Cache frequently accessed repos"? So they can opt-in only if they want to speed up the repos they usually work with (e.g. the last 10 repos that are accessed more than 10 times).

Reply:
I haven't checked the case with "Load entire tree at once" 😅

You know it better than me so I'm all ears, providing users with more options to control is good UX too.